Source/CSharpForUE/TypeGenerator/CSClass.cpp

```diff
@@ -95,8 +108,18 @@ void UCSClass::ProcessOutParameters(FOutParmRec* OutParameters, uint8* ArgumentD
 	}
 }
 
-void UCSClass::InvokeManagedEvent(UObject* ObjectToInvokeOn, const UCSFunction* Function, TArray<uint8>& ArgumentData, RESULT_DECL)
+bool UCSClass::InvokeManagedEvent(UObject* ObjectToInvokeOn, const UCSFunction* Function, TArray<uint8>& ArgumentData, FString& ExceptionMessage, RESULT_DECL)
 {
 	const FGCHandle ManagedObjectHandle = FCSManager::Get().FindManagedObject(ObjectToInvokeOn);
-	FCSManagedCallbacks::ManagedCallbacks.InvokeManagedMethod(ManagedObjectHandle.GetHandle(), Function->GetManagedMethod(), ArgumentData.GetData(), RESULT_PARAM);
+	struct { TCHAR* Data = nullptr; int ArrayNum = 0; int ArrayMax = 0; } ExceptionText;
+	int ResultCode = FCSManagedCallbacks::ManagedCallbacks.InvokeManagedMethod(ManagedObjectHandle.GetHandle(), Function->GetManagedMethod(), ArgumentData.GetData(), RESULT_PARAM, &ExceptionText);
+	if (ResultCode != 0)
+	{
+		ExceptionMessage = FString(static_cast<TCHAR*>(ExceptionText.ArrayNum, ExceptionText.Data));
+		// Commented as causing crash - possibly due to StringMarshaller allocating on the COM heap rather than the normal heap?
+		// Do we need a better way of passing strings to unreal without causing memory leaks?
+		// FMemory::Free(ExceptionText.Data);
```

Comment:
As comment says this needs some thinking about. I'm not sure why the COM heap is being used instead of the normal heap, and I'm not sure how the StringMarshaller (and ArrayMarshaller) avoids memory leaks. Ideally we'd let C# free the memory since it's the one allocating it, but I'm not sure how to let it know when then memory is ready to be free'd without adding another interop call here.